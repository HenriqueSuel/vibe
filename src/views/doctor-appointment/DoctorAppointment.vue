<template>
  <section class="container">
    <h1>Filtrar</h1>
    <div class="card list-filter">
      <div class="flex list-filter__inputs">
        <date-picker
          type="date"
          range
          placeholder="Selecionar um Periodo"
          v-model="filter.period"
        ></date-picker>
        <input
          type="text"
          placeholder="Nome do Paciente"
          v-model="filter.name"
        />
      </div>
      <div class="list-filter__checkbox">
        <span v-for="(status, index) of filter.status" :key="index">
          <input
            type="checkbox"
            :name="status.name"
            :id="status.name"
            :value="true"
            v-model="status.value"
          />
          <label :for="status.name">{{ status.name }}</label>
        </span>
      </div>
      <div class="list-filter__button">
        <button class="button-primary" @click="search">Buscar</button>
      </div>
    </div>

    <section class="card">
      <div class="list-patient">
        <h1>Historico</h1>
        <ul class="list" v-for="(history, i) of listHistory" :key="i">
          <li class="flex">
            <div class="list-patient__info">
              Data: {{ history.data | date }} - Nome do Paciente:
              {{ history.nameUser }} - Status:
              {{ history.status | status }}
            </div>
            <div
              class="list-patient__action"
              v-if="history.status === 'AGENDADO'"
            >
              <a @click="absence(history)">Não compareçeu</a>
              <router-link class="link" :to="'/registro/' + history.id">
                Começar a consulta
              </router-link>
            </div>
          </li>
        </ul>
      </div>
    </section>
  </section>
</template>

<script>
import DatePicker from 'vue2-datepicker';
import { mapGetters } from 'vuex';

export default {
  name: 'DoctorAppoitment',
  components: {
    DatePicker,
  },
  data() {
    return {
      filter: {
        period: null,
        name: '',
        status: [
          {
            name: 'Falta',
            value: false,
          },
          {
            name: 'Cancelado',
            value: false,
          },
          {
            name: 'Agendado',
            value: false,
          },
          {
            name: 'Realizada',
            value: false,
          },
          {
            name: 'Concluido',
            value: false,
          },
        ],
      },
      listHistory: [],
    };
  },
  computed: {
    ...mapGetters('AppointmentStore', ['filterAppointment']),
  },
  methods: {
    search() {
      this.listHistory = this.filterAppointment(this.filter);
    },
    async absence(appointment) {
      await this.$store.dispatch('AppointmentStore/updateStatusAppointment', {
        ...appointment,
        status: 'FALTA',
      });
      this.listHistory = [];
    },
  },
};
</script>

<style scoped lang="scss">
.container {
  padding: 0 36px;
}
.list-filter {
  &__inputs {
    display: flex;
  }
  &__checkbox {
    align-items: center;
    margin-left: 8px;
    margin-top: 16px;
    label {
      margin: 0 8px 0 4px;
    }
  }
  &__button {
    text-align: right;
  }
}

@media only screen and (max-width: 750px) {
  .list-filter {
    &__checkbox {
      span {
        display: flex;
        align-items: center;
      }
    }
    &__inputs {
      flex-direction: column-reverse;
    }
  }
  .list-patient {
    .list {
      .flex {
        flex-direction: column;
        height: 100%;
      }
    }
    &__info {
      margin: 24px 0;
    }
    &__action {
      align-self: flex-start;
      display: flex;
      flex-direction: column;
    }
  }
}

.list-patient {
  .list li {
    cursor: default;
    flex-direction: row;
    justify-content: right;
    align-items: center;
  }
  &__info {
    flex-grow: 1;
  }
  &__action {
    a {
      cursor: pointer;
      margin-right: 8px;
      &:hover {
        color: #e72d51;
      }
    }
  }
}

.link {
  color: black;
  text-decoration: none;
}
</style>
