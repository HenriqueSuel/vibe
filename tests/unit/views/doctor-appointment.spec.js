import Vuex from 'vuex';
import { shallowMount, createLocalVue } from '@vue/test-utils';
import DoctorAppointment from '@/views/doctor-appointment/DoctorAppointment.vue';
import flushPromises from 'flush-promises';

const localVue = createLocalVue();

localVue.use(Vuex);

describe('DoctorAppointment.vue', () => {
  let actions;
  let getters;
  let store;

  beforeEach(() => {
    actions = {
      updateStatusAppointment: jest.fn().mockResolvedValue({ id: '1' }),
    };

    getters = {
      filterAppointment: jest.fn().mockResolvedValue({ date: new Date(), nameUser: 'henrique' }),
    };

    store = new Vuex.Store({
      state: {},
      modules: {
        AppointmentStore: {
          namespaced: true,
          actions,
          getters,
        },
        DoctorStore: {
          namespaced: true,
          actions,
        },
        GlobalStore: {
          namespaced: true,
          actions,
        },
      },
    });
  });
  it('should change the input to cpf', () => {
    const wrapper = shallowMount(DoctorAppointment, {
      stubs: [],
      localVue,
      store,
    });
    expect(wrapper.html()).toMatchSnapshot();
  });
  it('should return a schedule list', async () => {
    const wrapper = shallowMount(DoctorAppointment, {
      stubs: [],
      localVue,
      store,
      computed: {
        ...getters,
      },
    });

    wrapper.setData({
      filter: {
        period: new Date(),
        name: 'Henrique',
      },
    });

    const buttonSubmit = wrapper.find('button');
    buttonSubmit.trigger('click');
    await flushPromises();
    expect(getters.filterAppointment).toHaveBeenCalled();
  });

  it('should change the status', async () => {
    const wrapper = shallowMount(DoctorAppointment, {
      stubs: [],
      localVue,
      store,
      computed: {
        ...getters,
      },
    });

    wrapper.vm.absence({ id: '123' });
    await flushPromises();
    expect(actions.updateStatusAppointment).toHaveBeenCalled();
  });
});
