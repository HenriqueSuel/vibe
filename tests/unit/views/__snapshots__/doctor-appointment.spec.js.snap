// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DoctorAppointment.vue should change the input to cpf 1`] = `
<section class="container">
  <h1>Filtrar</h1>
  <div class="card list-filter">
    <div class="flex list-filter__inputs">
      <date-picker-stub defaultvalue="Sun Sep 27 2020 00:00:00 GMT-0300 (GMT-03:00)" disableddate="function _default() {
          return false;
        }" type="date" getclasses="function _default() {
          return [];
        }" titleformat="YYYY-MM-DD" format="YYYY-MM-DD" timetitleformat="YYYY-MM-DD" disabledtime="function _default() {
          return false;
        }" hourstep="1" minutestep="1" secondstep="1" scrollduration="100" valuetype="date" range="true" rangeseparator=" ~ " placeholder="Selecionar um Periodo" editable="true" clearable="true" prefixclass="mx" inputclass="mx-input" inputattr="[object Object]" appendtobody="true" popupstyle="[object Object]" confirmtext="OK" shortcuts=""></date-picker-stub> <input type="text" placeholder="Nome do Paciente">
    </div>
    <div class="list-filter__checkbox"><span><input type="checkbox" name="Falta" id="Falta" value="true"> <label for="Falta">Falta</label></span><span><input type="checkbox" name="Cancelado" id="Cancelado" value="true"> <label for="Cancelado">Cancelado</label></span><span><input type="checkbox" name="Agendado" id="Agendado" value="true"> <label for="Agendado">Agendado</label></span><span><input type="checkbox" name="Realizada" id="Realizada" value="true"> <label for="Realizada">Realizada</label></span><span><input type="checkbox" name="Concluido" id="Concluido" value="true"> <label for="Concluido">Concluido</label></span></div>
    <div class="list-filter__button"><button class="button-primary">Buscar</button></div>
  </div>
  <section class="card">
    <div class="list-patient">
      <h1>Historico</h1>
    </div>
  </section>
</section>
`;
